purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1545,16 +1545,16 @@ class Purchases internal constructor(
         billing.findPurchaseInPurchaseHistory(
             appUserID,
             storeProduct.type,
-            upgradeInfo.oldSku,
+            upgradeInfo.oldProductId,
```

Comment:
Shouldn't this use `oldSku` or `oldProductId` - I'm a bit confused with the deprecated field. It looks like oldProductId doesn't have to be set always

Reply:
i'm confused about your comment, can you clarify? why does it look like it doesn't have to be set?

Reply:
Well.. if `UpgradeInfo` has two fields and people are only  setting `oldSku`, how is `oldProductInfo` being filled in? (This point is mute if we just rename instead of deprecating)

Reply:
ah i see, well it's a required field, the others have default values but oldProductId doesn't.

but either way i'm good w the rename

Reply:
Hmm I think we should deprecate instead of renaming... We are already forcing devs to do a lot of changes on this release, and I think we should minimize these required changes whenever possible. As for implementation details, I think we would need to:
- Change the `UpgradeInfo` data class to receive `oldProductId` (it's already done here)
- Add a deprecated computed property `oldSku` that just references `oldProductId`
- Add a deprecated secondary constructor that receives `oldSku` but actually sets `oldProductId`.

Reply:
Ohh you're completely right, I didn't consider that... Without it we would break it for devs using named parameters... So I guess we can just rename it as you had it then 👍

Reply:
🥴 got it, will undo the redo lol